_docs/master_agent.md

```diff
@@ -0,0 +1,120 @@
+---
+layout: architecture
+title: Master of Masters
+date: 2019-07-09
+version: v0.0.1
+summary: A complete master/agent architecture with multiple compile masters and load balancing for redundancy.
+---
+
+## Intended Audience
+
+This architecture is intended for large infrastructures or dynamic infrastructures
+that require the redundancy of multiple compile masters.
+
+
+<div class="mermaid">
+  graph LR;
+  git(Git Repository)
+  Foreman(The Foreman)
+  Webhook(Puppet Webhook Server)
+  AllCM((All Compile Masters))
+
+  PuppetDB
+
+  PuppetServerMoM{Master of Masters}
+
+  subgraph CM[Compile Masters]
+      Compile1[/Compile Master\]
+      Compile2[/Compile Master\]
+      Compile3[/Compile Master\]
+  end
+
+  LoadBalancer[Load Balancer]
+
+  Agent1(Agent 1)
+  Agent2(Agent 2)
+  Agents(Agents ...)
+  Agent_n(Agent n)
+
+  click Foreman "https://www.theforeman.org" "Foreman is a complete lifecycle management tool for physical and virtual servers."
+  click Webhook "https://github.com/voxpupuli/puppet_webhook" "A webhook service that can trigger code deploys from source code repository updates."
+
+  git --webhook--> Webhook
+  Webhook --r10k code deploy--> PuppetServerMoM
+  Webhook -.r10k code deploy.-> AllCM
+
+  PuppetDB --- PuppetServerMoM
+  Foreman --- PuppetServerMoM
+
+  PuppetServerMoM --> Compile1
+  PuppetServerMoM --> Compile2
+  PuppetServerMoM --> Compile3
+
+  Compile1 --> LoadBalancer
+  Compile2 --> LoadBalancer
+  Compile3 --> LoadBalancer
+
+  LoadBalancer --> Agent1
+  LoadBalancer --> Agent2
+  LoadBalancer --> Agents
+  LoadBalancer --> Agent_n
+</div>
+
+## Setup and Usage
+
+{write a guide on how to deploy, configure, and use this architecture}
+
+
+### Git Repository
+
+We recommend organizing your code as a Control Repository with branches for
+environments. See the [reference repository](https://github.com/puppetlabs/control-repo)
+for an example.
+
+
+### Foreman
+
+[Foreman](https://www.theforeman.org) is a complete lifecycle management tool
+for physical and virtual servers. It will provide you with a graphical
+classifier, a Hiera data source, and report monitoring. It also includes the
+power to easily automate repetitive tasks, quickly deploy applications, and
+proactively manage servers, on-premise or in the cloud.
+
+
+### Puppet Webhook
+
```

Comment:
In this section we probably should point to puppet module [puppetwebhook](https://github.com/voxpupuli/puppet-puppetwebhook/.)